server/controllers/courseController.mjs

```diff
@@ -51,6 +55,37 @@ export const createCourse = async (req, res) => {
   }
 };
 
+//fetch teacherId from course table and teacher data from user table
+export const getTeacherData = async (req, res) => {
+  const { uid } = req.params;
+
+  try {
+    const courseDoc = await db.collection(COURSES).doc(uid).get();
+
+    if (!courseDoc.exists) {
+      return res.status(404).send({ message: 'Teacher ID not found' });
+    }
+
+    const courseData = courseDoc.data(); // Get the entire document data
+    const teacherId = courseData.teacherId; // Extract the teacherId field
```

Comment:
Here you can use `createdBy` attribute and take from it `teacherEmail`. Then you can find user by the email in the line 72.

Reply:
teacherId is unique; it's the UID in users table. I think it's always best practice to find any record by UID. Let me know if I'm understanding this correctly.